### PR TITLE
fix typehints

### PR DESCRIPTION
**Pull request by JaskRendix**
continuing #1211 and separate from #1642
from
Found 452 errors in 90 files (checked 288 source files)
to
Found 336 errors in 69 files (checked 288 source files)

there are others, but this is the first package with the less tricky.

there are a bunch in the journal, etc, but the majority is always related to the translate component (41 in total), it's necessary for moving the widget, I have already asked on pygame_menu about.

---

regarding **tuxemon/states/monster/__init__.py**
before it was 6.5, but I tracked the value and it was always interpreted as 6.
Reason why I put 6 and we got rid of 2 typehints in one shot.

### CODE REVIEW

File: tuxemon/event/actions/open_shop.py [hunk 1]
```diff
@@ -89,4 +88,4 @@ def sell_menu() -> None:
         elif menu == "sell":
             push_sell_menu()
         else:
-            assert_never(menu)
```
Comment (kerizane) [inline tuxemon/event/actions/open_shop.py:92]: Why did you remove this

File: tuxemon/formula.py [hunk 2]
```diff
@@ -54,13 +55,15 @@ def simple_damage_multiplier(
...
         for target_type in target_types:
-            body = TYPES.get(target_type, TYPES["aether"])
-            if body.extra_damage is None:
-                continue
-            if attack_type == body.extra_damage:
-                m *= 2
-            elif attack_type == body.resist_damage:
-                m /= 2.0
+            if target_type:
+                assert target_type
```
Comment (kerizane) [inline tuxemon/formula.py]: this assert shouldn't be needed

File: tuxemon/menu/quantity.py [hunk 3]
```diff
@@ -160,9 +162,12 @@ def initialize_items(self) -> Generator[MenuItem[None], None, None]:
         count_len = 3
 
         cost = self.cost if self.quantity == 0 else self.quantity * self.cost
+        cost_tag: Union[str, int]
         if int(cost) == 0:
-            cost = T.translate("shop_buy_free")
+            cost_tag = T.translate("shop_buy_free")
+        else:
+            cost_tag = cost
```
Comment (kerizane) [inline tuxemon/menu/quantity.py]: Can we just do 

```
        if int(cost) == 0:
            cost_tag = T.translate("shop_buy_free")
        else:
            cost_tag = str(cost)
```

File: tuxemon/monster.py [hunk 4]
```diff
@@ -322,12 +322,16 @@ def load_from_db(self, slug: str) -> None:
...
 
         # Look up the monster's sprite image paths
-        self.front_battle_sprite = self.get_sprite_path(
-            results.sprites.battle1
-        )
-        self.back_battle_sprite = self.get_sprite_path(results.sprites.battle2)
-        self.menu_sprite_1 = self.get_sprite_path(results.sprites.menu1)
-        self.menu_sprite_2 = self.get_sprite_path(results.sprites.menu2)
+        if results.sprites:
+            assert results.sprites
```
Comment (kerizane) [inline tuxemon/monster.py]: Don't need this assert

File: tuxemon/states/pc_locker/__init__.py [hunk 5]
```diff
@@ -441,21 +444,21 @@ def find_monster_box(itm: Item, box: list) -> Optional[Item]:
...
-                        player.remove_item(itm)
-                    else:
-                        stored = player.find_item_in_storage(
-                            find_monster_box(itm, box).instance_id
-                        )
-                        stored.quantity += quantity
-                        itm.quantity = diff
+                retrieve = find_monster_box(itm, box)
+                if retrieve is not None:
+                    assert retrieve
```
Comment (kerizane) [inline tuxemon/states/pc_locker/__init__.py]: Not going to point this out any more. Please check all your asserts.

Comment (kerizane) [reply tuxemon/states/pc_locker/__init__.py]: The if check covers this, so the assert is pointless

File: tuxemon/states/phone/__init__.py [hunk 6]
```diff
@@ -474,8 +473,9 @@ def add_menu_items(
             fix_width(menu._width, -0.20), fix_height(menu._height, -0.03)
         )
 
+        widgets
```
Comment (kerizane) [inline tuxemon/states/phone/__init__.py]: ?

File: tuxemon/states/world/world_menus.py [hunk 7]
```diff
@@ -103,6 +106,7 @@ def monster_menu_hook() -> None:
                 # get the newly selected item.  it will be set to previous
                 # position
                 original_monster = monster_menu.get_selected_item().game_object
+                assert original_monster
```
Comment (kerizane) [inline tuxemon/states/world/world_menus.py:108]: This one is fine :)

**Review by kerizane — CHANGES_REQUESTED** [review]
Yay typehints :)

File: tuxemon/states/world/world_menus.py (hunk 7 above)
Comment (JaskRendix) [reply tuxemon/states/world/world_menus.py:108]: ok

File: tuxemon/states/phone/__init__.py (hunk 6 above)
Comment (JaskRendix) [reply tuxemon/states/phone/__init__.py]: forgot to remove

File: tuxemon/states/pc_locker/__init__.py (hunk 5 above)
Comment (JaskRendix) [reply tuxemon/states/pc_locker/__init__.py]: you're right, I went in goblin mode with asserts, I'm going to fix it

File: tuxemon/formula.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/formula.py]: ok

File: tuxemon/menu/quantity.py (hunk 3 above)
Comment (JaskRendix) [reply tuxemon/menu/quantity.py]: done

File: tuxemon/event/actions/open_shop.py (hunk 1 above)
Comment (JaskRendix) [reply tuxemon/event/actions/open_shop.py:92]: because of:
`tuxemon/event/actions/open_shop.py:92: error: Argument 1 to "assert_never" has incompatible type "str"; expected "NoReturn"`

Comment (JaskRendix) [reply tuxemon/event/actions/open_shop.py:92]: if you have another idea, then feel free

Comment (kerizane) [reply tuxemon/event/actions/open_shop.py:92]: Ok, let's raise an exception.

Comment (JaskRendix) [reply tuxemon/event/actions/open_shop.py:92]: ok

Comment (kerizane) [reply tuxemon/event/actions/open_shop.py:92]: Please provide a useful error msg

**Review by kerizane — CHANGES_REQUESTED** [review]

File: tuxemon/event/actions/open_shop.py (hunk 1 above)
Comment (JaskRendix) [reply tuxemon/event/actions/open_shop.py:92]: `"The parameter {self.menu} can be only 'both', 'buy' or 'sell'."`

**Review by kerizane — APPROVED** [review]
